sbt-plugin/src/main/scala/akka/grpc/sbt/AkkaGrpcPlugin.scala

```diff
@@ -117,41 +137,39 @@ object AkkaGrpcPlugin extends AutoPlugin {
     generators.map { generator =>
       protocbridge.Target(generator, targetPath, generator match {
-        case PB.gens.java => settings.filterNot(_ == "flat_package")
-        case _ => settings
+        case PB.gens.java => baseSettings.filterNot(_ == "flat_package")
+        case _ => baseSettings
       })
     }
   }
 
   // creates a seq of generator and per generator settings
-  def generatorsFor(stubs: Seq[AkkaGrpc.GeneratedSource], languages: Seq[AkkaGrpc.Language], scalaBinaryVersion: ScalaBinaryVersion, logger: GenLogger): Seq[protocbridge.Generator] = {
+  def generatorsFor(stubs: Seq[AkkaGrpc.GeneratedSource], languages: Seq[AkkaGrpc.Language], options: Seq[String], scalaBinaryVersion: ScalaBinaryVersion, logger: GenLogger): Seq[protocbridge.Generator] = {
+    import AkkaGrpc._
     // these two are the model/message (protoc) generators
     def ScalaGenerator: protocbridge.Generator = protocbridge.JvmGenerator("scala", ScalaPbCodeGenerator)
     // we have a default flat_package, but that doesn't play with the java generator (it fails)
     def JavaGenerator: protocbridge.Generator = PB.gens.java
 
-    stubs match {
-      case Seq(_, _) =>
-        languages match {
-          case Seq(_, _) => Seq(ScalaGenerator, toGenerator(ScalaBothCodeGenerator, scalaBinaryVersion, logger), JavaGenerator, toGenerator(JavaBothCodeGenerator, scalaBinaryVersion, logger))
-          case Seq(AkkaGrpc.Scala) => Seq(ScalaGenerator, toGenerator(ScalaBothCodeGenerator, scalaBinaryVersion, logger))
-          case Seq(AkkaGrpc.Java) => Seq(JavaGenerator, toGenerator(JavaBothCodeGenerator, scalaBinaryVersion, logger))
-        }
-      case Seq(AkkaGrpc.Client) =>
-        languages match {
-          case Seq(_, _) => Seq(ScalaGenerator, toGenerator(ScalaClientCodeGenerator, scalaBinaryVersion, logger), JavaGenerator, toGenerator(JavaClientCodeGenerator, scalaBinaryVersion, logger))
-          case Seq(AkkaGrpc.Scala) => Seq(ScalaGenerator, toGenerator(ScalaClientCodeGenerator, scalaBinaryVersion, logger))
-          case Seq(AkkaGrpc.Java) => Seq(JavaGenerator, toGenerator(JavaClientCodeGenerator, scalaBinaryVersion, logger))
-        }
-      case Seq(AkkaGrpc.Server) =>
-        languages match {
-          case Seq(_, _) => Seq(ScalaGenerator, toGenerator(ScalaServerCodeGenerator, scalaBinaryVersion, logger), JavaGenerator, toGenerator(JavaServerCodeGenerator, scalaBinaryVersion, logger))
-          case Seq(AkkaGrpc.Scala) => Seq(ScalaGenerator, toGenerator(ScalaServerCodeGenerator, scalaBinaryVersion, logger))
-          case Seq(AkkaGrpc.Java) => Seq(JavaGenerator, toGenerator(JavaServerCodeGenerator, scalaBinaryVersion, logger))
-        }
-    }
+    lazy val serverPowerApis = options.contains(GeneratorOption.ServerPowerApis.setting)
+    lazy val baseGenerators: Seq[Generator] = Seq(ScalaGenerator, toGenerator(ScalaTraitCodeGenerator, scalaBinaryVersion, logger))
```

Comment:
Having these separate seems fine by me. It's hard to say whether this will introduce complexity in the Maven and Gradle plug-ins, but we'll find out when those are enabled again.